src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1981,14 +2058,31 @@ internal void Add(object input)
                 return;
             }
 
-            if (_inputFormat == NativeCommandIOFormat.Text)
+            if (_inputFormat is not NativeCommandIOFormat.Text)
             {
-                AddTextInput(input);
+                AddXmlInput(input);
+                return;
             }
-            else // Xml
+
+            object baseObjInput = input;
+            if (input is PSObject pso)
             {
-                AddXmlInput(input);
+                baseObjInput = pso.BaseObject;
             }
+
+            if (baseObjInput is byte[] bytes)
+            {
+                _streamWriter.BaseStream.Write(bytes, 0, bytes.Length);
+                return;
+            }
+
+            if (baseObjInput is byte b)
+            {
+                _streamWriter.BaseStream.WriteByte(b);
+                return;
+            }
+
+            AddTextInput(input);
```

Comment:
This bit enables piping bytes directly to native commands. Might need to be in a different PR, not strictly in scope.